Secret-agency-web/src/main/react/secret-agency/src/components/reports-page/ReportsPage.tsx

```diff
@@ -51,6 +55,45 @@ export class ReportsPage extends React.PureComponent<IProps, IState> {
         this.setState(prevState => ({...prevState, navigateToDetailId: reportId, navigateToDetailMode:"EDIT"}))
     };
 
+    private onApproveReport = async (reportId: number): Promise<void> => {
+        const result = await reportService.approveReport(reportId);
+        // error
+        if (typeof result === "string") {
+            this.setState(prevState => ({...prevState, serverError: result}));
+            return;
+        }
+
+        // success
+        this.updateReportStatus(reportId, "APPROVED");
```

Comment:
constanty by byly safer

Reply:
Souhlasím. 😄 Ale ten parametr té funkce je typovaný, takže by to neplatný string snad nahlásilo.